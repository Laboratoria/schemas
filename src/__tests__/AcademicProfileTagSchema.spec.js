const mongoose = require('mongoose/browser');
const { AcademicProfileTagSchema, UserSchema, TagSchema } = require('../../')(mongoose);

describe('AcademicProfileTagSchema', () => {
  it('should fail validation when fields missing', () => {
    const doc = new mongoose.Document({}, AcademicProfileTagSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should validate example', () => {
    const academicProfileTag = new mongoose.Document({}, AcademicProfileTagSchema);
    const createBy = new mongoose.Document({}, UserSchema);
    const tag = new mongoose.Document({}, TagSchema);

    const academicProfileTagSchema = new mongoose.Document({
      academicProfileTag: academicProfileTag._id,
      createBy: createBy._id,
      tag: tag._id,
      reason: 'foo',
    }, AcademicProfileTagSchema);

    academicProfileTagSchema.validate();
  });
});
