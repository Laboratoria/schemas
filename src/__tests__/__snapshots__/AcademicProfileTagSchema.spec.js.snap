// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcademicProfileTagSchema should fail validation when fields missing 1`] = `
Object {
  "academicProfile": [ValidatorError: Path \`academicProfile\` is required.],
  "createdBy": [ValidatorError: Path \`createdBy\` is required.],
  "tag": [ValidatorError: Path \`tag\` is required.],
}
`;
